Goobi/src/de/sub/goobi/metadaten/Metadaten.java

```diff
@@ -1067,141 +1071,123 @@ public String KnotenDelete() throws IOException {
-                parent.removeChild(iter.next());
-            }
-
-            /* anschliessend die neue Childliste anlegen */
-            for (Iterator<DocStruct> iter = alleDS.iterator(); iter.hasNext();) {
-                parent.addChild(iter.next());
-            }
-        }
-
-        /*
-         * -------------------------------- als erstes Child --------------------------------
-         */
-        if (this.neuesElementWohin.equals("3")) {
-            DocStructType dst = this.myPrefs.getDocStrctTypeByName(this.addDocStructType2);
-            ds = this.mydocument.createDocStruct(dst);
-            DocStruct parent = this.myDocStruct;
-            if (parent == null) {
-                logger.debug("das gewählte Element kann den Vater nicht ermitteln");
-                return "Metadaten3links";
-            }
-            List<DocStruct> alleDS = new ArrayList<DocStruct>();
-            alleDS.add(ds);
+                // Build a new list of children for the edited element
+                List<DocStruct> newChildren = new ArrayList<>(edited.getAllChildren().size() + 1);
+                if (mode.equals(TreeInsertionMode.AS_FIRST_CHILD)) {
+                    for (DocStruct createdElement : createdElements) {
+                        selection.addChild(createdElement);
+                    }
+                }
+                for (DocStruct child : edited.getAllChildren()) {
+                    if (child == selection && mode.equals(TreeInsertionMode.BEFORE_ELEMENT)) {
+                        for (DocStruct element : createdElements) {
+                            selection.addChild(element);
+                        }
+                    }
+                    newChildren.add(child);
+                    if (child == selection && mode.equals(TreeInsertionMode.AFTER_ELEMENT)) {
+                        for (DocStruct element : createdElements) {
+                            selection.addChild(element);
+                        }
+                    }
+                }
 
-            if (parent.getAllChildren() != null && parent.getAllChildren().size() != 0) {
-                alleDS.addAll(parent.getAllChildren());
-                parent.getAllChildren().retainAll(new ArrayList<DocStruct>());
-            }
+                // Remove the existing children
+                for (DocStruct child : newChildren) {
+                    edited.removeChild(child);
+                }
 
-            /* anschliessend die neue Childliste anlegen */
-            for (Iterator<DocStruct> iter = alleDS.iterator(); iter.hasNext();) {
-                parent.addChild(iter.next());
+                // Set the new children on the edited element
+                for (DocStruct child : newChildren) {
+                    edited.addChild(child);
+                }
             }
+            return createdElements.iterator().next();
+        } catch (UGHException iek) {
+            throw new RuntimeException(iek.getMessage(), iek);
```

Comment:
It will be shown as a stack trace the same for checked and unchecked exceptions. In my eyes there are some unfortunate design decisions in the Ugh and that UGHException and its subclasses are _checked_ exceptions is one of them. If _docstruct has no type_ ever happens the user cannot handle it anyway, it’s a programming error then.